tests/unit/s2n_server_signature_algorithms_extension_test.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+#include <stdint.h>
+
+#include "tls/s2n_config.h"
+#include "tls/s2n_connection.h"
+#include "tls/extensions/s2n_server_signature_algorithms.h"
+#include "tls/s2n_tls.h"
+#include "tls/s2n_tls13.h"
+
+#include "stuffer/s2n_stuffer.h"
+#include "utils/s2n_safety.h"
+
+int main(int argc, char **argv)
```

Comment:
One of the main benefits of unit tests is that you can test failure cases that would be difficult to hit with integ tests. You might want to add a few quick tests for malformed messages to make sure they're handled right.

Reply:
Added more tests to s2n_tls13_cert_request_test.c working on s2n_server_signature_algorithms_extension_test.c